spec/consensus/light-client.md

```diff
@@ -1,318 +1,577 @@
 
-- header fields
+ ```go
+   type Header struct {
+        Height   int64
+        Time     Time      // the chain time when the header (block) was generated
 
-  - _height_
-  - _bfttime_: the chain time when the header (block) was generated
-  - _V_: validator set containing validators for this block.
-  - _NextV_: validator set for next block.
-  - _commit_: evidence that block with height _height_ - 1 was committed by a set of validators (canonical commit). We will use `signers(commit)` to refer to the set of validators that committed the block.
+        // hashes from the app output from the prev block
+        ValidatorsHash     []byte        // hash of the validators for the current block
+        NextValidatorsHash []byte        // hash of the validators for the next block
 
-- signed header fields: contains a header and a _commit_ for the current header; a "seen commit". In the Tendermint consensus the "canonical commit" is stored in header _height_ + 1.
+        // hashes of block data
+       	LastCommitHash []byte            // hash of the commit from validators from the last block
+   }
 
-- For each header _h_ it has locally stored, the lite client stores whether
-  it trusts _h_. We write _trust(h) = true_, if this is the case.
+   type SignedHeader struct {
+        Header        Header
+        Commit        Commit            // commit for the given header
+   }
 
-- Validator fields. We will write a validator as a tuple _(v,p)_ such that
-  - _v_ is the identifier (we assume identifiers are unique in each validator set)
-  - _p_ is its voting powers
+   type ValidatorSet struct {
+        Validators         []Validator
+        TotalVotingPower   int64
+   }
 
-### Functions
+   type Validator struct {
+        Address       Address           // validator address (we assume validator's addresses are unique)
+        VotingPower   int64             // validator's voting power
+   }
 
-For the purpose of this lite client specification, we assume that the Tendermint Full Node exposes the following function over Tendermint RPC:
+   type TrustedState {
+        SignedHeader   SignedHeader
+        ValidatorSet   ValidatorSet
+   }
+ ```
 
+### Functions
+
+For the purpose of this lite client specification, we assume that the Tendermint Full Node exposes the following functions over Tendermint RPC:
 ```go
+    // returns signed header: Header with Commit, for the given height
     func Commit(height int64) (SignedHeader, error)
-      // returns signed header: header (with the fields from
-      // above) with Commit that include signatures of
-      // validators that signed the header
-
 
-    type SignedHeader struct {
-      Header        Header
-      Commit        Commit
-    }
+    // returns validator set for the given height
+    func Validators(height int64) (ValidatorSet, error)
 ```
 
-### Definitions
+Furthermore, we assume the following auxiliary functions:
```

Comment:
It might be helpful if we distinguish here between the functions that are necessary for implementation and those that are just necessary for defining the model/stating invariants (eg. validators(hash) is just for stating the failure model)